app/enterprise.go

```diff
@@ -202,6 +205,6 @@ func (a *App) initEnterprise() {
 		})
 	}
 	if cloudInterface != nil {
-		a.srv.Cloud = cloudInterface(a)
+		a.srv.Cloud = cloudInterface(a.Srv(), a.Srv().Store, a.HTTPService())
```

Comment:
I would suggest to move it to the initEnterprise of the server, not the the App one.